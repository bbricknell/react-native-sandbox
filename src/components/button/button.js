import React, { Component } from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';

export class Button extends Component {
  render() {
    return(
      <TouchableOpacity style={styles.btn} onPress={this.props.clickAction}>
        <Text style={styles.text}>{this.props.text}</Text>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  btn: {
    minWidth: 100,
    minHeight: 50,
    marginVertical: 10,
    padding: 10,
    borderWidth: 2.5,
    borderRadius: 50,
    borderColor: '#5F9EA0'
  },
  text: {
    textAlign: 'center',
    fontSize: 18,
    fontWeight: 'bold',
    color: '#5F9EA0'
  }
});