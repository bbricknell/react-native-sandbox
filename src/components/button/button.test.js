import React from 'react';
import Renderer from 'react-test-renderer';

import { Button } from './button';

describe('Button Component', () => {
  test('renders button correctly', () => {
    const button = Renderer.create(<Button text="button test" clickAction={() => {}} />);
    expect(button).toMatchSnapshot();
  });
});