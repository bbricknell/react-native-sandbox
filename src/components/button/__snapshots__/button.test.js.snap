// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders button correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "#5F9EA0",
      "borderRadius": 50,
      "borderWidth": 2.5,
      "marginVertical": 10,
      "minHeight": 50,
      "minWidth": 100,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#5F9EA0",
        "fontSize": 18,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    button test
  </Text>
</View>
`;
